internal/txapp/txapp.go

```diff
@@ -560,34 +560,41 @@ func (r *TxApp) ProposerTxs(ctx context.Context, txNonce uint64, maxTxsSize int6
 		events = events[:50]
 	}
 
-	// Final events are the events whose bodies have not been received by the network
-	var finalEvents []*types.VotableEvent
+	ids := make([]types.UUID, 0, len(events))
 	for _, event := range events {
-		// Check if the event body is already received by the network
-		containsBody, err := resolutionContainsBody(ctx, readTx, event.ID())
-		if err != nil {
-			return nil, err
-		}
-		if containsBody {
-			continue
-		}
+		ids = append(ids, event.ID())
+	}
 
-		finished, err := isProcessed(ctx, readTx, event.ID())
-		if err != nil {
-			return nil, err
-		}
-		if finished {
-			continue
+	doesNotHaveBody, err := voting.FilterExistsNoBody(ctx, readTx, ids...)
+	if err != nil {
+		return nil, err
+	}
+
+	notProcessed, err := voting.FilterNotProcessed(ctx, readTx, doesNotHaveBody...)
+	if err != nil {
+		return nil, err
+	}
+
+	eventMap := make(map[types.UUID]*types.VotableEvent)
+	for _, evt := range events {
+		eventMap[evt.ID()] = evt
+	}
+
+	finalEvents := make([]*types.VotableEvent, 0)
+	for _, id := range notProcessed {
+		event, ok := eventMap[id]
+		if !ok {
+			// this should never happen
+			return nil, fmt.Errorf("internal bug: event with id %s not found", id.String())
 		}
 
-		// MaxTxBytes restrictions per block are enforced here
-		// As the rlp encoded size is almost always smaller, enforcing it on the unencoded data for ease of use
 		evtSz := int64(len(event.Type)) + int64(len(event.Body))
 		if evtSz > maxTxsSize {
 			break
 		}
-		finalEvents = append(finalEvents, event)
 		maxTxsSize -= evtSz
+		maxTxsSize -= rlpEncodingOverheadSize
```

Comment:
If we are doing this for every event, we should probably reduce the OverheadSize to something proportional to size. Because right now its defined to be quite high upto 1000.

Reply:
Hmm I see. I think we should do it per event, since it will be constant. Let me try smthn